static_frame/core/index_hierarchy.py

```diff
@@ -1565,6 +1568,37 @@ def to_pandas(self) -> 'DataFrame':
         mi.names = self.names
         return mi
 
+    def to_tree(self) -> TreeNodeT:
+        '''Returns the tree representation of an IndexHierarchy
+        '''
+        def add_labels(tree: TreeNodeGrowableT, labels: tp.Sequence[tp.Any]) -> None:
+            # For a set of labels, add them into a tree
+            outermost_label, *inner_labels = labels
+            if len(inner_labels) == 1:
+                if outermost_label not in tree:
+                    tree[outermost_label] = []
+                tree[outermost_label].append(inner_labels[0])
+                return
+
+            if outermost_label not in tree:
+                tree[outermost_label] = {}
+            add_labels(tree[outermost_label], inner_labels)
+
+        tree: TreeNodeGrowableT = {}
+        for labels in self.iter_label():
+            add_labels(tree, labels)
+
+        def clean(tree: TreeNodeGrowableT) -> None:
+            # Make the leafs immutable!
+            for k in tuple(tree.keys()):
+                if isinstance(tree[k], list):
+                    tree[k] = tuple(tree[k])
+                else:
+                    clean(tree[k])
```

Comment:
As the tree is just a temporary container between index objects, do we need to do a full-depth transversal to convert `list` to `tuple`? Or, can it be done in the initial traversal?

Reply:
Probably -> but that would require me to know how `IndexLevels` work...

The current algorithm iterates through each label of the index hierarchy, so I don't have access to the innermost index objects.

Let me take a look to see if I how to iterate more intelligently.